src/main/java/LinkedBinarySearchTree.java

```diff
@@ -0,0 +1,492 @@
+            return current.getData();
+        } else {
+            return max(current.getRight());
+        }
+    }
+
+    @Override
+    public boolean contains(T element) {
+        return binarySearch(element, root);
+    }
+
+    /**
+     * Helper method enabling a recursive binary search for a given element.
+     *
+     * @param element Element being searched for
+     * @param current Current binary search tree node being investigated
+     * @return True if the element is found, false otherwise
+     */
+    private boolean binarySearch(T element, Node<T> current) {
+        if (current == null) {
+            return false;
+        } else if (current.getData().equals(element)) {
+            return true;
+        } else {
+            if (current.getData().compareTo(element) > 0) {
+                return binarySearch(element, current.getLeft());
+            } else {
+                return binarySearch(element, current.getRight());
+            }
+        }
+    }
+
+    @Override
+    public int getCount(T element) {
+        if (isEmpty()) {
+            return 0;
+        }
+        return getCount(element, root);
+    }
+
+    /**
+     * Helper method for recursive count of elements in binary search tree.
+     * Remember, equal elements are added to the right subtree.
+     *
+     * @param element Element to be counted
+     * @param current Current node being investigated
+     * @return Number of times the element exists in the (sub)tree.
+     */
+    private int getCount(T element, Node<T> current) {
+        if (current == null) {
+            return 0;
+        }
+        if (current.getData().equals(element)) {
+            return 1 + getCount(element, current.getRight());
+        } else if (current.getData().compareTo(element) > 0) {
+            return getCount(element, current.getLeft());
+        } else {
+            return getCount(element, current.getRight());
+        }
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size() == 0;
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    /**
+     * Recursive pre order traversal of a binary tree. Elements are added
+     * to a bag in the required order.
+     *
+     * @param current  current node in the tree being referenced
+     * @param sequence Bag containing elements in pre order order
+     */
+    private void preOrder(Node<T> current, IndexedBag<T> sequence) {
+        if (current != null) {
+            sequence.add(current.getData());
+            preOrder(current.getLeft(), sequence);
+            preOrder(current.getRight(), sequence);
+        }
+    }
+
+    /**
+     * Recursive in order traversal of a binary tree. Elements are added
+     * to a bag in the required order.
+     *
+     * @param current  current node in the tree being referenced
+     * @param sequence Bag containing elements in in order order
+     */
+    private void inOrder(Node<T> current, IndexedBag<T> sequence) {
```

Comment:
Use of the `Bag` feels a little like cheating to me. Copying the entire tree to another data structure then iterating that seems to miss the mark.

Reply:
For this I was considering an aside. Getting this deep into trees is already perhaps a tad _pushing the envelope_ already for a CS102 course. 

That said, totally worth including because it's cool for them to use a Stack inside the iterator for multiple reasons, but because I want them to see the other bits of the tree as the "important" (for lack of a better term) bits, having the binary tree iterator as an aside may help highlight things better.